core/src/com/unciv/ui/newgamescreen/GameOptionsTable.kt

```diff
@@ -71,7 +72,10 @@ class GameOptionsTable(
         checkboxTable.addOneCityChallengeCheckbox()
         checkboxTable.addNuclearWeaponsCheckbox()
         checkboxTable.addIsOnlineMultiplayerCheckbox()
-        checkboxTable.addReligionCheckbox(cityStateSlider)
+        if (gameParameters.baseRuleset != BaseRuleset.Civ_V_Vanilla.fullName)
```

Comment:
If you go this way, I'd check whether either the starting era or the mod have a unique disabling religion rather than checking specifically for the vanilla ruleset